app/utils.py

```diff
@@ -154,29 +162,141 @@ def setup_logger(name, level=logging.INFO):
     return logger
 
 
-def validate_resource(json):
-    validation_errors = {'errors': {'type': 'validation'}}
+def validate_resource(request, id=-1):
+    json = request.get_json()
+    validation_errors = {'errors': [],
+                         'type': 'validation'}
+
     if not json:
-        message = "A JSON body is required to use this endpoint, but none was given"
-        validation_errors['errors']['message'] = message
+        message = generate_error(ValidationError.MissingJSONError, field=None)
```

Comment:
Nit: is message still a good name for this considering it's now a dict with a bunch of data in it? Maybe we should call this `error`